libraries/shared/storage/utils/decoder_test.go

```diff
@@ -146,5 +146,29 @@ var _ = Describe("Storage decoder", func() {
 			Expect(decodedValues[0]).To(Equal(big.NewInt(0).SetBytes(common.HexToHash("AB54A98CEB1F0AD2").Bytes()).String()))
 			Expect(decodedValues[1]).To(Equal(big.NewInt(0).SetBytes(common.HexToHash("38D7EA4C67FF8E502B6730000").Bytes()).String()))
 		})
+
+		It("decodes address + 2 uint48s", func() {
+			//TODO: replace with real data when available
+			addressHex := "0000000000000000000000000000000000012345"
+			packedStorage := common.HexToHash("00000002a300" + "000000002a30" + addressHex)
```

Comment:
Could you remind me why the order in the hash here is uint, uint, address but the `packedTypes` and `decodedValues` maps are ordered address, uint, uint?

Reply:
Yeah I also found that weird. Based on the actual storage diffs we have for the packed `ttl` + `tau` storage slots in the auction contracts, it looks like Solidity packs values 'backwards' from the way you'd expect. I.e. even though `ttl` comes before `tau` in the contract storage, the diff value hash has the `tau` value first. So when I fudged this diff, I figured that since `guy` comes first in the contract storage, it will be last in the packed hash. Looking forward to confirming that once we get some real bid storage diffs though